functions/AutoRemediateS3-002.js

```diff
@@ -0,0 +1,78 @@
+"use strict";
+
+const config = require('./config');
+const AWS = require("aws-sdk");
+
+const CCRuleCode = 'S3-002'
+const CCRuleName = 'BucketPublicReadAcpAccess'
+const allUsersURI = 'http://acs.amazonaws.com/groups/global/AllUsers'
+const readAcpPermission = "READ_ACP"
+
+module.exports = {
+
+  remediateAllUsers: function (thisGrant, newAcl) {
+    if (thisGrant.Permission != readAcpPermission) {  // any besides READ are passed through
+      newAcl['Grants'].push(thisGrant);
+    }
+
+    return newAcl;
+  },
+
+  transferOwner: function (oldAcl, newAcl) {
+    newAcl.Owner = oldAcl.Owner; // transfer the existing bucket owner
+
+    return newAcl;
+  },
+
+  transferAcl: function (oldAcl, newAcl) {
+    this.transferOwner(oldAcl, newAcl);
+
+    // now, act on any grants to all users - and just copy over any other grants
+    oldAcl.Grants.forEach(function (grant, i) { if (grant.Grantee.URI == allUsersURI) { this.remediateAllUsers(grant, newAcl) } else { newAcl['Grants'].push(grant) }; })
```

Comment:
Rather than a `forEach`, I think a more _functional_ approach would be to `filter` out the grants you don't want

`var validGrants = oldAcl.Grants.filter( validGrants )`

where:
```
var ruleViolation = (grant) => grant.Grantee.URI === allUsersURI && thisGrant.Permission == readAcpPermission

var validGrants = (grant) => !ruleViolation(grant)
```

It should make it easier to test, since each function can be tested independently